xbmc/pvr/channels/PVRChannel.cpp

```diff
@@ -187,6 +189,12 @@ bool CPVRChannel::UpdateFromClient(const std::shared_ptr<CPVRChannel>& channel)
   SetClientID(channel->ClientID());
   SetArchive(channel->HasArchive());
 
+  if (m_iClientProviderUid != channel->ClientProviderUid())
+  {
+    m_iClientProviderUid = channel->ClientProviderUid();
+    m_bChanged = true;
+  }
+
```

Comment:
Would it make sense to encapsulate this into a member function `SetClientProviderUid()` ?

Reply:
Sure, a private member function as this should really never be called elsewhere.

Reply:
Done